### PR TITLE
Feature/MemoryGameCard

### LINKED ISSUE #244: <MemoryGameCard /> - Lesson
## The story:
As a **user** that **wants to test my memory**, I want to **play memory game by matching "cards" in a lesson**

---

## What we should do:
We should **build a "flippable" card** so that **the user can play memory games**.

---

## To be successful we need to:
_Check these off as they are completed:_
- [ ] Card should be hidden by default
- [ ] User can click on a card to flip it and reveal the answer (animated in css)
- [ ] Should take prop `flipped` as a boolean
- [ ] Should take prop `state` as either `'correct'` OR `'incorrect`'

---

## Design
Default: 
image

Flipped:
image

Correct:
image

Incorrect:
image

[LINK TO FIGMA DESIGN](https://figma.com)

---

## Notes
This ticket **DOES NOT** include the grid, the game or the lesson. It's JUST the **individual** `<MemoryCard />` component

### PR DESCRIPTION
**Pull request by Ellie-2020**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Fixed issues (PRs without this will not be accepted)
- Fixes #244 

## Changes
- add MemoryGameCard component to Client component
- add checkCircle icon to UI Icon

## PR Checklist
- [ ] Does your PR contain relevant tests that pass?
- [x] Is your code linted? `yarn lint` or `yarn lint:fix`
- [x] Does your code build correctly in ALL projects? `yarn build`

### CODE REVIEW

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+export interface MemoryGameCardProps extends CardProps {
+    flipped?: boolean;
+    state?: 'correct' | 'incorrect';
+    content?: string;
+}
+
+const StyledCard = styled(Card)<MemoryGameCardProps>`
+  position: relative;
+  width: 160px;
+  height: 160px;
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: Can we please use a `${theme.size()}` value here, or at least a rem? 🙏

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  left: 0%;
+  top: 0%;
+
+  background: ${t.color('white')};
+  box-shadow: 0px 2px 10px ${t.color('grey.200')};
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: We should have a `t.shadow` available for this. If it's not configured for what we need, feel free to add it too!

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  border-radius: ${t.borderRadius.large};
+
+  align-items: center;
+
+  & > span {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: For this, we can use the `centerAbsolute` found [here](https://github.com/code-mentoring/learn/blob/f1bdcab02b0b5308a786914abdd64abc72b76eda/packages/ui/css/common.ts#L3)

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    display: flex;
+    text-align: center;
+    font-family: Source Code Pro;
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: This should come form the theme

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  // there is no corlor as GrayBlue/400, use grey.400 instead
+  ${p => p.flipped && `
+    color: ${t.color('grey.400')};
+  `};
+
+  ${p => p.state === 'incorrect' && `
+    border: 2px solid ${t.color('tertiary.500')};
+    color: ${t.color('primary.600')};
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: Remove this. We'll keep it the same color :)

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  `};
+
+  ${p => p.state === 'correct' && `
+    width: 140px;
+    height: 140px;
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: Can we please use a `transform: scale()` here instead? That way we can also animate it :)

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: Can be removed

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    background: ${t.color('secondary.100')};
+    opacity: 0.2;
+    border: 2px solid ${t.color('secondary.400')};
+    color: ${t.color('primary.600')};
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: Same here. We'll keep it the same color 🙏

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  `};
+`;
+
+const StyledIcon = styled(Icon)`
+  position: absolute;
+  top: 3.32%;
+  right: 3.32%;
```
Comment (tristanMatthias) [inline packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: Can we use `theme.size('tiny')` for this please?

**Review by tristanMatthias — CHANGES_REQUESTED** [review]

**Comment by tristanMatthias** [comment]
Hey @Ellie-2020! Thanks a lot for getting started on this one 😄. Got a number of requests, sorry if it's a lot lol.

One thing we will definitely need is a "flipping" animation. Have a google around for a "css card flip" there are lots of tutorials. If you are having trouble still, let me know and we can jump on a call and do it together. Maybe even make a webinar out of it!

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx (hunk 1 above)
Comment (Ellie-2020) [reply packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: ${theme.size('missive')} is 7.2rem which is not big enough. So changed to 16rem.

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx (hunk 2 above)
Comment (Ellie-2020) [reply packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: `<Card /> `has shadow and background setup already, so remove shadow and background from here.

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx (hunk 3 above)
Comment (Ellie-2020) [reply packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: cool! changed

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx (hunk 4 above)
Comment (Ellie-2020) [reply packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: add 'code' to fontFamily in theme.
```
  fontFamily: {
    ...
    code: ['"Source Code Pro"', 'Open Sans', 'helvetica', 'sans-serif'].join(', ')
  },
```

File: packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx (hunk 6 above)
Comment (Ellie-2020) [reply packages/client/src/components/MemoryGameCard/MemoryGameCard.tsx]: cool! changed.

**Comment by Ellie-2020** [comment]
I have added the flipping animation.

**Comment by tristanMatthias** [comment]
Hey @Ellie-2020! Great work! I made a few adjustments, hope that's ok.
1. Moved styles out to their own file
2. Simplified component structure a bit to just use one child element
3. Moved `content` prop to just `children`
4. Small style optimizaions for flip/transitions.

Thanks a lot for submitting this. Looks great!

**Review by tristanMatthias — APPROVED** [review]

**Pull request merged**